pkg/controller/servicemesh/controlplane/reconciler.go

```diff
@@ -416,6 +417,17 @@ func (r *ControlPlaneReconciler) validateSMCPSpec(spec v1.ControlPlaneSpec) erro
 func (r *ControlPlaneReconciler) renderCharts() error {
 	//Generate the spec
 	r.Status.LastAppliedConfiguration = r.Instance.Spec
+	if len(r.Status.LastAppliedConfiguration.Version) == 0 {
+		// initialize version
+		if len(r.Status.AppliedVersion) == 0 {
+			// this must be from a 1.0 operator
+			r.Status.LastAppliedConfiguration.Version = common.LegacyMaistraVersion
```

Comment:
...And we're only using the legacy (`v1.0`) version when `Status.AppliedVersion` is not set.